src/BaGet.Hosting/FileProvider/BaGetSpaStaticFileProvider.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.IO;
+using BaGet.Core;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.SpaServices.StaticFiles;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace BaGet.Hosting
+{
+    public class BaGetSpaStaticFileProvider : ISpaStaticFileProvider
+    {
+        public BaGetSpaStaticFileProvider(IWebHostEnvironment environment, IOptions<BaGetOptions> options, ILogger<BaGetSpaStaticFileProvider> logger)
+        {
+            var rootPath = "BaGet.UI/build";
+            if (environment.IsDevelopment())
+            {
+                rootPath = "../" + rootPath;
+            }
+
+            var combinedRootPath = new DirectoryInfo(Path.Combine(environment.ContentRootPath, rootPath));
+            if (combinedRootPath.Exists)
+            {
+                var pathBase = options.Value.PathBase;
```

Comment:
You can modify BaGet's configs as the app is running and it will automatically refresh them. This is capturing the `PathBase` at startup. Can we pass the options down to the file provider and let that read the latest `PathBase` each time it serves a file?